### DISCUSSION TITLE
[nx-flutter] reading pubspec

### DISCUSSION THREAD

**Discussion by ghost** [tinesoft/nxrocks#25] [Ideas]
could the plugin read apps and library pubspecs to determine dependencies graph, as it does in js/ts by analysing imports  ? that would be awesome

**Comment by tinesoft** [comment]
Hi,

Great idea 💡 This could be a very valuable addition to the plugin indeed.
I don't know yet how Nx does it for TS/JS, I will need to dig into that.

 In the meantime,  you are more than welcome if you want to submit a PR for this! 😉

I've created #28 to track that

**Comment by ghost** [comment]
I'd love to help, and would have too if I wasn't a complete noob in js/ts, don't know where to start

I've read about automatic dependencies [here](https://nx.dev/latest/angular/workspace/structure/dependency-graph#how-the-dependency-graph-is-built) , seems they parse all imports ?

I've hardcoded (implicit) dependencies atm (parent-child in the config file, easy), and I can see the nx dependency graph alright:

<img width="1751" alt="Nx_Workspace_Dependency_Graph_-_Vivaldi" src="https://user-images.githubusercontent.com/52695909/105683905-79987d80-5ef4-11eb-9e16-5a680e3e5be6.png">

colleagues of mine are using nx for js apps, I'll ask around

**Comment by asadkhanek** [comment]
Hi,

Great idea 💡 This could be a very valuable addition to the plugin indeed.
I don't know of Nx does it for TS/JS, I will need to dig into that.

In the meantime, you are more than welcome if you want to submit a PR for this! 😉

I've created #28 to track that

**Comment by asadkhanek** [comment]
Great idea 💡 This could be a very valuable addition to the plugin indeed.

**Comment by tinesoft** [comment]
Hi @benoitskipr , @asadkhanek  and future googlers

The Nx dependency graph support has landed in the plugin since https://github.com/tinesoft/nxrocks/releases/tag/nx-flutter%2Fv2.0.0 ! 

Enjoy!
